src/arc.rs

```diff
@@ -171,6 +177,118 @@ fn rotate_pt(pt: Vec2, angle: f64) -> Vec2 {
     )
 }
 
+impl ParamCurve for Arc {
+    fn eval(&self, t: f64) -> Point {
+        let angle = self.start_angle + (self.sweep_angle * t);
+        sample_ellipse(self.radii, self.x_rotation, angle).to_point()
+    }
+
+    fn subsegment(&self, range: Range<f64>) -> Self {
+        Self {
+            center: self.center,
+            radii: self.radii,
+            start_angle: self.start_angle + (self.sweep_angle * range.start),
+            sweep_angle: self.sweep_angle - (self.sweep_angle * (range.end - range.start)),
+            x_rotation: self.x_rotation,
+        }
+    }
+
+    fn start(&self) -> Point {
+        sample_ellipse(self.radii, self.x_rotation, self.start_angle).to_point()
+    }
+
+    fn end(&self) -> Point {
+        sample_ellipse(
+            self.radii,
+            self.x_rotation,
+            self.start_angle + self.sweep_angle,
+        )
+        .to_point()
+    }
+}
+
+impl ParamCurveArclen for Arc {
+    fn arclen(&self, accuracy: f64) -> f64 {
+        // Normalize ellipse to have radius y >= radius x, required for the parameter assumptions
+        // of `incomplete_elliptic_integral_second_kind`.
+        let (radii, mut start_angle) = if self.radii.y >= self.radii.x {
+            (self.radii, self.start_angle)
+        } else {
+            (
+                Vec2::new(self.radii.y, self.radii.x),
+                self.start_angle + PI / 2.,
+            )
+        };
+        let m = 1. - (radii.x / radii.y).powi(2);
+
+        // Normalize sweep angle to be non-negative
+        let mut sweep_angle = self.sweep_angle;
+        if sweep_angle < 0. {
+            start_angle = -start_angle;
+            sweep_angle = -sweep_angle;
+        }
+
+        // Normalize start angle to be on the upper half of the ellipse
+        let start_angle = start_angle.rem_euclid(PI);
+        let end_angle = start_angle + sweep_angle;
+
+        let mut quarter_turns = (2. / PI * end_angle).trunc() - (2. / PI * start_angle).trunc();
+        let end_angle = end_angle % PI;
+
+        // The elliptic arc length is equal to radii.y * (E(end_angle | m) - E(start_angle | m))
+        // with E the incomplete elliptic integral of the second kind and parameter
+        // m = 1 - (radii.x / radii.y)^2 = k^2.
+        //
+        // See also:
+        // https://en.wikipedia.org/w/index.php?title=Ellipse&oldid=1248023575#Arc_length
+        //
+        // The implementation here allows calculating the incomplete elliptic integral in the range
+        // 0 <= phi <= 1/2 pi (the first elliptic quadrant), so split the arc into segments in
+        // that range.
+        let mut arclen = 0.;
+
+        // The available accuracy (tolerance) is distributed over the calculation of the two
+        // incomplete and one complete elliptic integrals.
```

Comment:
The distribution of the accuracy tolerance is not very clever, but I'd rather not spend too many clock cycles on it, when those cycles could be spent iterating the convergence loops.

The worst case is when the number of complete quarter turns, evaluated as complete elliptic integrals, is very high, as the tolerance available to it decreases linearly in the number of quarter turns.